bsdlib/doc/gnss_extension.rst

```diff
@@ -145,61 +150,61 @@ The elevation mask parameter controls the elevation threshold angle (in degrees)
- |
- | **Caution:** The Altitude in meter has to have an accuracy in the range of +/- 10’s of meter of your actual height. An erroneous altitude will result in a quite sever error in 3 satellite position fix calculation.
+.. important::
+   The altitude must be accurate to a value within ±10 meters of the actual altitude of the device. An erroneous altitude will result in a severe error in the position fix calculation using three satellites.
 
 .. code-block:: c
 
-    typedef struct
-    {
-        int32_t latitude          = 0;     // Not known, set to 0
-        int32_t longitude         = 0;     // Not known, set to 0
-        int16_t altitude          = meter; // Your actual height in meter
-        uint8_t unc_semimajor     = 127;   // Both set to 127 when latitude
-        uint8_t unc_semiminor     = 127;   // and longitude are unknown
-        uint8_t orientation_major = 0;     // Not known, set to 0
-        uint8_t unc_altitude      = 0;     // Set to 0
-        uint8_t confidence        = 100;   // Set to 100
-    } nrf_gnss_agps_data_location_t;
-
-If the actual altitude of where the device is located changes vs the altitude stored in the
-GNSS engine (e.g. the device is moving around), the accuracy of the 3 sat fix will be degraded.
-
-This Low Accuracy mode is not the same as the 2D fix you see documented in the NMEA reports.
-All fixes, including the Low Accuracy fix, will be reported as 3D fix.
-To get information of how many satellites are used for fix look at the $GPGSA report and number
-of IDs of SVs used in position fix (not set equal to zero).
+   nrf_gnss_agps_data_location_t location;
+
+   location.latitude          = 0;        /* Unknown, set to 0 */
+   location.longitude         = 0;        /* Unknown, set to 0 */
+   location.altitude          = altitude; /* Actual altitude of the device in meters */
+   location.unc_semimajor     = 127;      /* Set to 127 when location is unknown */
+   location.unc_semiminor     = 127;      /* Set to 127 when location is unknown */
+   location.orientation_major = 0;        /* Unknown, set to 0 */
+   location.unc_altitude      = 0;        /* Set to 0 */
+   location.confidence        = 100;      /* Set to 100 */
+
+If the actual altitude of the device changes with respect to the altitude stored in the GNSS engine (for example, when the device moves around), the accuracy of the position fix using three satellites will be degraded.
+
+The low accuracy mode is different from the 2D fix that is documented in the NMEA reports.
+All fixes, including the low accuracy fixes, will be reported as 3D fixes.
+See the $GPGSA report and number of IDs of SVs used in the position fix to get information of the number of satellites are used for the position fix.
+Only the SV IDs which are not empty are valid.
```

Comment:
I think the line `Only the SV IDs which are not empty are valid.` could be removed. The satellites used in a fix can be checked from $GPGSA with a 3D fix, i.e. a line which starts with `$GPGSA,A,3`. The previous line could be changed for example like this:

To see the number of satellites used in a fix, check the $GPGSA line starting with $GPGSA,A,3 for a list of SVs used for that fix.

Reply:
My thinking why I wrote it like I did was: you will always get "A,3" regarddles of if this was a 3, 4, or 5 sat fix, but by refering like I did you know explicitly if this was a 3 sat fix or if it has been able to track more than 3 sat.

Reply:
@bjorn-tore-taraldsen The number of satellites listed in the $GPGSA message tell how many satellites were used in a fix and I think it should be self explanatory that empty fields do not count as used satellites, but this is just my opinion. For example:

3 sat fix:
`$GPGSA,A,3,01,03,22,,,,,,,,,,5.25,4.46,2.76,1*18`
4 sat fix:
`$GPGSA,A,3,01,03,22,31,,,,,,,,,5.25,4.46,2.76,1*18`
5 sat fix:
`$GPGSA,A,3,01,03,22,31,32,,,,,,,,5.25,4.46,2.76,1*18`

Reply:
Had a chat with Tommi and agreed that we just remove the last sentence "Only the SV IDs which are not empty are valid" and let the rest stay as is.

Reply:
Just as side note, why do we discuss NMEA strings in this document which is not enabled by default? Would it make much more sense to discuss the default format which is also much more practical to process programmatically?